### PR TITLE
COOK-63 Append -Dhdp.version=#{hdp_version} to Java options

### PR DESCRIPTION
**Pull request by wolf31o2**
Needed for CDAP-2881 testing via Coopr...

### CODE REVIEW

File: attributes/config.rb [hunk 1]
```diff
@@ -44,3 +44,25 @@
...
+    '2.2.4.4-16'
+  when '2.2.6.0'
+    '2.2.6.0-2800'
+  else
+    node['hadoop']['distribution_version']
+  end
+
+if node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f >= 2.2 && node['cdap']['version'].to_f >= 3.1
+  default['cdap']['cdap_env']['opts'] = "${OPTS} -Dhdp.version=#{hdp_version}"
+  default['cdap']['cdap_site']['app.program.jvm.opts'] = "-XX:MaxPermSize=128M ${twill.jvm.gc.opts} -Dhdp.version=#{hdp_version}"
```
Comment (jwang47) [inline attributes/config.rb]: This should probably be documented in the CDAP docs too. Could you create a JIRA for that?

File: attributes/config.rb [hunk 2]
```diff
@@ -44,3 +44,25 @@
   default['cdap']['cdap_site']['gateway.server.port'] = '10000'
   default['cdap']['cdap_site']['gateway.memory.mb'] = '512'
 end
+
+# HDP 2.2+ support
+hdp_version =
+  case node['hadoop']['distribution_version']
+  when '2.2.0.0'
+    '2.2.0.0-2041'
```
Comment (jwang47) [inline attributes/config.rb]: Do we know what the numbers after the `-` mean?

File: attributes/config.rb (hunk 1 above)
Comment (wolf31o2) [inline attributes/config.rb]: https://issues.cask.co/browse/CDAP-2891

File: attributes/config.rb (hunk 2 above)
Comment (wolf31o2) [inline attributes/config.rb]: Yeah.. they're iterations. They're internal to HDP and aren't really documented, anywhere. I pull them from the directories on-disk after installing or from the package names...

```
# yum list installed | grep hadoop | grep -v ambari | head -n 1
hadoop_2_2_6_0_2800.x86_64               2.6.0.2.2.6.0-2800.el6        @HDP-2.2
```

The first three parts are the actual Hadoop version, 2.6.0... the rest is the HDP version and matches for all packages in a given HDP release.

**Comment by jwang47** [comment]
LGTM

**Pull request merged**